vividus-plugin-applitools/src/main/java/org/vividus/visual/eyes/converter/ExamplesTableToApplitoolsVisualChecksConverter.java

```diff
@@ -0,0 +1,213 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.visual.eyes.converter;
+
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import com.applitools.eyes.BatchInfo;
+import com.applitools.eyes.MatchLevel;
+import com.applitools.eyes.RectangleSize;
+import com.applitools.eyes.config.Configuration;
+
+import org.apache.commons.lang3.Validate;
+import org.apache.commons.lang3.reflect.TypeLiteral;
+import org.jbehave.core.model.ExamplesTable;
+import org.jbehave.core.steps.ParameterConverters.AbstractParameterConverter;
+import org.vividus.ui.action.search.Locator;
+import org.vividus.ui.screenshot.ScreenshotConfiguration;
+import org.vividus.ui.screenshot.ScreenshotParameters;
+import org.vividus.ui.screenshot.ScreenshotParametersFactory;
+import org.vividus.visual.eyes.factory.ApplitoolsVisualCheckFactory;
+import org.vividus.visual.eyes.model.ApplitoolsVisualCheck;
+import org.vividus.visual.model.VisualActionType;
+import org.vividus.visual.screenshot.BaselineIndexer;
+
+public class ExamplesTableToApplitoolsVisualChecksConverter extends
+        AbstractParameterConverter<ExamplesTable, List<ApplitoolsVisualCheck>> implements ApplitoolsVisualCheckFactory
+{
+    private static final String BASELINE_NAME_FIELD = "baselineName";
+    private static final String ACTION_FIELD = "action";
+    private static final String EXECUTE_API_KEY_FIELD = "executeApiKey";
+    private static final String READ_API_KEY_FIELD = "readApiKey";
+    private static final String HOST_APP_FIELD = "hostApp";
+    private static final String HOST_OS_FIELD = "hostOS";
+    private static final String VIEWPORT_SIZE_FIELD = "viewportSize";
+    private static final String MATCH_LEVEL_FIELD = "matchLevel";
+    private static final String SERVER_URI_FIELD = "serverUri";
+    private static final String APP_NAME_FIELD = "appName";
+    private static final String BATCH_NAME_FIELD = "batchName";
+    private static final String BASELINE_ENV_NAME_FIELD = "baselineEnvName";
+    private static final String ELEMENTS_TO_IGNORE_FIELD = "elementsToIgnore";
+    private static final String AREAS_TO_IGNORE_FIELD = "areasToIgnore";
+
+    private static final List<String> SUPPORTED_FIELDS = List.of(BASELINE_NAME_FIELD, ACTION_FIELD,
+            EXECUTE_API_KEY_FIELD, READ_API_KEY_FIELD, HOST_APP_FIELD, HOST_OS_FIELD, VIEWPORT_SIZE_FIELD,
+            MATCH_LEVEL_FIELD, SERVER_URI_FIELD, APP_NAME_FIELD, BATCH_NAME_FIELD, BASELINE_ENV_NAME_FIELD,
+            ELEMENTS_TO_IGNORE_FIELD, AREAS_TO_IGNORE_FIELD);
+
+    private String executeApiKey;
+    private String readApiKey;
+    private String hostApp;
+    private String hostOS;
+    private RectangleSize viewportSize;
```

Comment:
```suggestion
    private Dimension viewportSize;
```
I still recommend to keep `Dimension` type here